src/Api/EmailValidationV4.php

```diff
@@ -0,0 +1,282 @@
+     *
+     * @return mixed|ResponseInterface
+     *
+     * @throws Exception
+     */
+    public function createBulkPreview(string $previewId, $filePath)
+    {
+        Assert::stringNotEmpty($previewId);
+
+        if (strlen($filePath) < PHP_MAXPATHLEN && is_file($filePath)) {
+            $fileData = ['filePath' => $filePath];
+        } else {
+            $fileData = [
+                'fileContent' => $filePath,
+                'filename' => 'file',
+            ];
+        }
+
+        $postDataMultipart = [];
+        $postDataMultipart[] = $this->prepareFile('file', $fileData);
+
+        $response = $this->httpPostRaw(sprintf('/v4/address/validate/preview/%s', $previewId), $postDataMultipart);
+        $this->closeResources($postDataMultipart);
+
+        return $this->hydrateResponse($response, CreateBulkPreviewResponse::class);
+    }
+
+    /**
+     * @param string $previewId ID given when the list created
+     *
+     * @return mixed|ResponseInterface
+     *
+     * @throws Exception
+     */
+    public function getBulkPreview(string $previewId)
+    {
+        Assert::stringNotEmpty($previewId);
+
+        $response = $this->httpGet(sprintf('/v4/address/validate/preview/%s', $previewId));
+
+        return $this->hydrateResponse($response, GetBulkPreviewResponse::class);
+    }
+
+    /**
+     * @param string $previewId ID given when the list created
+     *
+     * @return bool
+     */
+    public function deleteBulkPreview(string $previewId)
+    {
+        Assert::stringNotEmpty($previewId);
+
+        $response = $this->httpDelete(sprintf('/v4/address/validate/preview/%s', $previewId));
+
+        return 204 === $response->getStatusCode();
+    }
+
+    /**
+     * @param string $previewId ID given when the list created
+     *
+     * @return mixed|ResponseInterface
+     *
+     * @throws Exception
+     */
+    public function promoteBulkPreview(string $previewId)
+    {
+        Assert::stringNotEmpty($previewId);
+
+        $response = $this->httpPut(sprintf('/v4/address/validate/preview/%s', $previewId));
+
+        return $this->hydrateResponse($response, PromoteBulkPreviewResponse::class);
+    }
+
+    /**
+     * @param string $fieldName Field Name
+     * @param array  $filePath  ['fileContent' => 'content'] or ['filePath' => '/foo/bar']
+     *
+     * @return array File Data
+     */
+    private function prepareFile(string $fieldName, array $filePath): array
+    {
+        $filename = isset($filePath['filename']) ? $filePath['filename'] : null;
+
+        if (isset($filePath['fileContent'])) {
+            // File from memory
+            $resource = fopen('php://temp', 'r+');
+            fwrite($resource, $filePath['fileContent']);
+            rewind($resource);
+        } elseif (isset($filePath['filePath'])) {
+            // File form path
+            $path = $filePath['filePath'];
+
+            // Remove leading @ symbol
+            if (0 === strpos($path, '@')) {
```

Comment:
What will happen if we have 2x `@`? Maybe is safer running a regular expression with `preg_replace`?

![image](https://user-images.githubusercontent.com/7809429/104486684-032a8000-55c4-11eb-956b-6463cb951757.png)

Reply:
@DavidGarciaCat So this is needed when you have a file [file.txt], try to set filepath as:
'@file.txt' like here: https://github.com/mailgun/mailgun-php/blob/master/tests/Message/MessageBuilderTest.php#L146

and this condition will remove the leading zero, the same condition exists here: https://github.com/mailgun/mailgun-php/blob/master/src/Api/Message.php#L141

Reply:
I think this should be removed. It is a legacy layer for new code.

If we fear that some will use a file path like `@/foo/bar.txt` then we should update the documentation.